packages/pluginutils/src/composable-filters.ts

```diff
@@ -148,6 +160,26 @@ export function exclude(expr: FilterExpression): Exclude {
   return new Exclude(expr);
 }
 
+/**
+ * convert a queryObject to FilterExpression like
+ * ```js
+ *   and(query(k1, v1), query(k2, v2))
+ * ```
+ * @param queryFilterObject The query filter object needs to be matched.
+ * @returns a `And` FilterExpression
+ *
+ * There are three kinds of conditions are supported:
+ * 1. `boolean`: if the value is `true`, the key must exist and be truthy. if the value is `false`, the key must not exist or be falsy.
+ * 2. `string`: the key must exist and be equal to the value.
+ * 3. `RegExp`: the key must exist and match the value.
+ */
+export function queryObjectToFilterExpr(queryFilter: QueryFilterObject): And {
```

Comment:
This helper function could be used for some simply scenario like https://github.com/vitejs/vite-plugin-vue/blob/main/packages/plugin-vue/src/index.ts#L362